From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Sing-Han Chen <singhanc@nvidia.com>
Date: Mon, 27 May 2024 10:21:00 +0000
Subject: [PATCH 143/241] NVIDIA: SAUCE: usb: gadget: udc: Add Tegra264 support

This commit adds support for XUSB device mode controller support on
Tegra264 SoC. The configuration is mostly copied from Tegra234 XUDC.

U1/U2 support is disabled in initial configuration.

http://nvbugs/4295138

Change-Id: I35c7a3a4803e8cf0556bf467d47c47524abec3dd
Signed-off-by: Sing-Han Chen <singhanc@nvidia.com>
(cherry picked from commit 38f94bc0658880dc8c7c679e52828f6e0544100e)
Signed-off-by: Bodla Rakesh Babu <rbodla@nvidia.com>

diff --git a/drivers/usb/gadget/udc/tegra-xudc.c b/drivers/usb/gadget/udc/tegra-xudc.c
index 17c30e21b357..83ab8082010f 100644
--- a/drivers/usb/gadget/udc/tegra-xudc.c
+++ b/drivers/usb/gadget/udc/tegra-xudc.c
@@ -3747,6 +3747,20 @@ static struct tegra_xudc_soc tegra234_hv_xudc_soc_data = {
 	.is_hv_guest = true,
 };
 
+static struct tegra_xudc_soc tegra264_xudc_soc_data = {
+	.clock_names = tegra186_xudc_clock_names,
+	.num_clks = ARRAY_SIZE(tegra186_xudc_clock_names),
+	.num_phys = 4,
+	.u1_enable = false,
+	.u2_enable = false,
+	.lpm_enable = true,
+	.invalid_seq_num = false,
+	.pls_quirk = false,
+	.port_reset_quirk = false,
+	.has_ipfs = false,
+	.has_pg_support = true,
+};
+
 static const struct of_device_id tegra_xudc_of_match[] = {
 	{
 		.compatible = "nvidia,tegra210-xudc",
@@ -3768,6 +3782,10 @@ static const struct of_device_id tegra_xudc_of_match[] = {
 		.compatible = "nvidia,tegra234-hv-xudc",
 		.data = &tegra234_hv_xudc_soc_data
 	},
+	{
+		.compatible = "nvidia,tegra264-xudc",
+		.data = &tegra264_xudc_soc_data
+	},
 	{ }
 };
 MODULE_DEVICE_TABLE(of, tegra_xudc_of_match);
-- 
2.43.0

